src/emulator/util/src/util.cpp

```diff
@@ -184,3 +185,37 @@ std::string toupper(const std::string &s) {
 }
 
 } // namespace string_utils
+
+template <>
+std::uint16_t byte_swap(std::uint16_t val) {
+    return (val >> 8) | (val << 8);
+}
+
+template <>
+std::uint32_t byte_swap(std::uint32_t val) {
+    //        AA             BB00                   CC0000                  DD000000
+    return (val >> 24) | (val >> 8) & 0xFF00 | (val << 8) & 0xFF0000 | (val << 24) & 0xFF000000;
+}
+
+template <>
+std::uint64_t byte_swap(std::uint64_t val) {
+    val = ((val << 8) & 0xFF00FF00FF00FF00ULL) | ((val >> 8) & 0x00FF00FF00FF00FFULL);
+    val = ((val << 16) & 0xFFFF0000FFFF0000ULL) | ((val >> 16) & 0x0000FFFF0000FFFFULL);
+
+    return (val << 32) | (val >> 32);
+}
```

Comment:
This is fine but wow there are a lot of numbers.

I was making a JRE a while back and I have this function from then:
```
        template <typename T>
        T swapEndian(T value) {
            T result = 0;
            for (int a = 0; a < sizeof(value); a++) {
                result <<= 8;
                result |= ((value >> (a * 8)) & 0xFF);
            }
            return result;
        }
```

It should work with any integer. Up to you if you want to use it. I think it does the same thing, but I am not totally sure.

Reply:
Thanks! ~~It's mine now~~

Reply:
I'm not sure now. I dont like loop

Reply:
That's fine.